src/control/server/config/server_test.go

```diff
@@ -568,7 +568,64 @@ func TestServerConfig_Validation(t *testing.T) {
 			},
 			expErr: FaultConfigInsufficientHugePages(4096, 2048),
 		},
-		"calculated hugepages": {
+		"insufficient hugepages set in config; emulated bdevs configured": {
+			extraConfig: func(c *Server) *Server {
+				return c.WithNrHugePages(2048).
+					WithEngines(
+						engine.NewConfig().
+							WithStorage(
+								storage.NewTierConfig().
+									WithStorageClass("ram").
+									WithScmRamdiskSize(1).
+									WithScmMountPoint("/foo"),
+								storage.NewTierConfig().
+									WithStorageClass("file").
+									WithBdevDeviceList("/tmp/daos-bdev").
+									WithBdevFileSize(16),
+							).
+							WithFabricInterfacePort(1234).
+							WithFabricInterface("eth0").
+							WithTargetCount(8).
+							WithPinnedNumaNode(0),
+					)
+			},
+			expErr: FaultConfigInsufficientHugePages(4096, 2048),
+		},
+		"insufficient hugepages set in config; no bdevs configured": {
+			extraConfig: func(c *Server) *Server {
+				return c.WithNrHugePages(2048).
+					WithEngines(
+						engine.NewConfig().
+							WithStorage(
+								storage.NewTierConfig().
+									WithStorageClass("ram").
+									WithScmRamdiskSize(1).
+									WithScmMountPoint("/foo"),
+							).
+							WithFabricInterfacePort(1234).
+							WithFabricInterface("eth0").
+							WithTargetCount(8).
+							WithPinnedNumaNode(0),
+					)
+			},
+			expConfig: baseCfg().
+				WithAccessPoints("hostname1:10001").
+				WithNrHugePages(2048).
+				WithEngines(
+					engine.NewConfig().
+						WithStorage(
+							storage.NewTierConfig().
+								WithStorageClass("ram").
+								WithScmRamdiskSize(1).
+								WithScmMountPoint("/foo"),
+						).
+						WithFabricInterfacePort(1234).
+						WithFabricInterface("eth0").
+						WithTargetCount(8).
+						WithPinnedNumaNode(0),
+				),
+		},
+		"zero hugepages set in config": {
```

Comment:
very minor, may not be worth it if the rest of the PR has no comments. Consider adding "; bdevs configured" to the name